### PR TITLE
[Tech debt] - Add dev env api endpoints

### PR DESCRIPTION
**Pull request by Heyrows**
⚠️  Disclaimer, the ACL provider to upload screenshots doesn't seems to be working but is not a main feature for running the E2E testing so it should be enough for now.

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/server/src/api/services/getEnv.ts [hunk 1]
```diff
@@ -39,17 +33,18 @@ function validate(config: Models.Manifest): ConfigErrors {
...
+    const url = new URL(apiEndpoint);
+    const host = url.host; // myrepo.prismic.io
+    const repository = host.split(".")[0]; // myrepo
+    return repository;
+  } catch (e) {
+    // Should already be covered by the start script.
+    console.error(
+      "It seems your repository API endpoint in your SM.json is wrong. Please start machine again to get instructions on how to fix it."
+    );
+    process.exit(0);
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/services/getEnv.ts:50]: pretty sure we don't want the server to stop running if the urls not valid

File: packages/client/src/models/ApisEndpoints.ts [hunk 2]
```diff
@@ -21,3 +23,33 @@ export const StageApisEndpoints: ApisEndpoints = {
...
+    console.error(`
+      It seems you're trying to use Slice Machine in DEV mode without all the environment variables
+      Here are the required environment variables
+      - Authentication API: ${authenticationApi ? "👍" : "⛔️"}
+      - Custom Type API: ${modelsApi ? "👍" : "⛔️"}
+      - Users API: ${usersApi ? "👍" : "⛔️"}
+      - Acl Provider API: ${aclApi ? "👍" : "⛔️"}
+    `);
+    // Stopping slice machine
+    process.exit(0);
```
Comment (MarcMcIntosh) [inline packages/client/src/models/ApisEndpoints.ts:45]: I guess we're the only ones that will to deal with this exiting

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]
are the modifications to the package-locks needed?

Edit: Yes, it's just the client's version that has been updated in the package locks.

File: packages/slice-machine/server/src/api/services/getEnv.ts (hunk 1 above)
Comment (Heyrows) [reply packages/slice-machine/server/src/api/services/getEnv.ts:50]: It's something only us would see, a user would have to have "wroom.test" in his API endpoint 😅.
i figured as devs, we might want to know as soon as possible if something is broken in our dev env.

What do you think should be the correct behaviour ?

Comment (MarcMcIntosh) [reply packages/slice-machine/server/src/api/services/getEnv.ts:50]: pretty sure etEnv is used in a few places, do we want it to exit if the apiEndpointIs wrong?

Comment (MarcMcIntosh) [reply packages/slice-machine/server/src/api/services/getEnv.ts:50]: I checked this an it looks like it should crash 
https://miro.com/app/board/uXjVOBPxAAY=/

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**